agent/os_helper.py

```diff
@@ -220,18 +222,13 @@ def auto_upgrades_enabled():
                 update_package_lists == '1' and \
                 '${distro_id}:${distro_codename}' in allowed_origins and \
                 '${distro_id}:${distro_codename}-security' in allowed_origins
-    else:
-        # For rpm-based distros.
+        return False
+    elif is_amazon_linux2():  # For Amazon Linux 2.
         # 1. check if yum-cron installed
         # 2. check if it's running
         # 3. check if it has proper values in its config file
         import rpm
-        try:
-            from sh import systemctl
-        except ImportError:
-            # No systemd - probably yum-cron is not running
-            # TODO: use "service" executable which also works without systemd and on older systems
-            return False
+        from sh import systemctl
```

Comment:
I wrote this so it wouldn't fail if running in container. Why did you change this?

Reply:
> I suggest renaming is_amazon2()

I don't agree. We can do that only after testing everything on RHEL and CentOS, not beforehand.

Reply:
you replied to a different comment